relayer/workers/parachaincommitmentrelayer/main.go

```diff
@@ -0,0 +1,120 @@
+package parachaincommitmentrelayer
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/sirupsen/logrus"
+	"github.com/snowfork/polkadot-ethereum/relayer/chain"
+	"github.com/snowfork/polkadot-ethereum/relayer/chain/ethereum"
+	"github.com/snowfork/polkadot-ethereum/relayer/chain/parachain"
+	"github.com/snowfork/polkadot-ethereum/relayer/contracts/inbound"
+	"github.com/snowfork/polkadot-ethereum/relayer/crypto/secp256k1"
+	"github.com/snowfork/polkadot-ethereum/relayer/substrate"
+)
+
+type Worker struct {
+	parachainConfig             *parachain.Config
+	ethereumConfig              *ethereum.Config
+	parachainConn               *parachain.Connection
+	parachainCommitmentListener *ParachainCommitmentListener
+	ethereumConn                *ethereum.Connection
+	ethereumChannelWriter       *EthereumChannelWriter
+	log                         *logrus.Entry
+}
+
+const Name = "parachain-commitment-relayer"
+
+func NewWorker(parachainConfig *parachain.Config, ethereumConfig *ethereum.Config) (*Worker, error) {
+	log := logrus.WithField("parachain-commitment-relayer", Name)
+
+	fmt.Println("Creating parachain-commitment-relayer")
+
+	ethereumKp, err := secp256k1.NewKeypairFromString(ethereumConfig.PrivateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	parachainConn := parachain.NewConnection(parachainConfig.Endpoint, nil, log)
+	ethereumConn := ethereum.NewConnection(ethereumConfig.Endpoint, ethereumKp, log)
+	// channel for messages from substrate
+	var subMessages = make(chan []chain.Message, 1)
```

Comment:
Why is this channel buffered?

Reply:
also ported from old code, likely not necessary - will address with other deadlock related fixes if needed